zebra-consensus/src/transaction.rs

```diff
@@ -195,11 +200,35 @@ where
 
                     if let Some(shielded_data) = shielded_data {
                         check::shielded_balances_match(&shielded_data, *value_balance)?;
+
                         for spend in shielded_data.spends() {
-                            // TODO: check that spend.cv and spend.rk are NOT of small
-                            // order.
+                            // Consensus rule: cv and rk MUST NOT be of small
+                            // order, i.e. [h_J]cv MUST NOT be 𝒪_J and [h_J]rk
+                            // MUST NOT be 𝒪_J.
+                            //
                             // https://zips.z.cash/protocol/protocol.pdf#spenddesc
+                            check::spend_cv_rk_not_small_order(spend)?;
+
+                            // Consensus rule: The proof π_ZKSpend MUST be valid
+                            // given a primary input formed from the other
+                            // fields except spendAuthSig.
+                            //
+                            // Queue the verification of the Groth16 spend proof
+                            // for each Spend description while adding the
+                            // resulting future to our collection of async
+                            // checks that (at a minimum) must pass for the
+                            // transaction to verify.
+                            let spend_rsp = spend_verifier
+                                .ready_and()
+                                .await?
+                                .call(primitives::groth16::ItemWrapper::from(spend).into());
```

Comment:
minor: we can probably change the `spend_verifier` so its request type is `Into<Item>` so that this into call happens automatically inside the service

Reply:
ooo neat